lib/active_record_host_pool/connection_proxy.rb

```diff
@@ -52,6 +53,20 @@ def send(symbol, *args, &blk)
     end
     ruby2_keywords :send if respond_to?(:ruby2_keywords, true)
 
+    def ==(other)
+      self.class == other.class &&
+        other.respond_to?(:unproxied) && @cx == other.unproxied &&
+        other.respond_to?(:database) && @database == other.database
```

Comment:
I don't think the respond_to check is necessary since you already checked the class

Reply:
The class check alone isn’t enough: https://github.com/zendesk/active_record_host_pool/blob/7b5ec2d9461a2ed46cb82c2addb9ee4c3d663a9e/lib/active_record_host_pool/connection_proxy.rb#L28-L31